src/shared/mount-util.c

```diff
@@ -543,12 +519,12 @@ static int mount_switch_root_move(const char *path) {
         return 0;
 }
 
-int mount_switch_root(const char *path, MountAttrPropagationType type) {
+int mount_switch_root(const char *path, unsigned long mount_propagation_flag) {
         _cleanup_close_ int fd_newroot = -EBADF;
-        unsigned long flags;
         int r;
 
         assert(path);
+        assert(IN_SET(mount_propagation_flag, 0, MS_SHARED, MS_SLAVE, MS_PRIVATE));
```

Comment:
we have this same test at one other place already, maybe we should add a helper for this, i.e.

```c
static inline MOUNT_PROPAGATION_FLAG_IS_VALID(unsigned long f) {
        return IN_SET(f, MS_SHARED, MS_SLAVE, MS_PRIVATE);
}
```

(could also be named lowercase. just named it uppercase because it has this macro-like feel, but doesn't matter)

(but anyway this doesn't really matter)

Reply:
Done.